minifi_main/JsonSchema.h

```diff
@@ -0,0 +1,26 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+
+namespace org::apache::nifi::minifi::docs {
+
+std::string generateJsonSchema();
```

Comment:
Would it be possible to add a test to check the json schema, with an attached json file that contains all the possible elements, to check if our generated schema is valid? It could be appended later if new elements are introduced and to update the test.

Reply:
The first result I found was https://github.com/pboettch/json-schema-validator which depends on nlohmann json library, maybe this could be used to test the generated schema.

Reply:
integrated the proposed validator, and added tests to verify that it reports errors on invalid configs

Reply:
Great addition, thank you!